src/worker.cpp

```diff
@@ -1185,4 +1184,281 @@ void Worker::collectFrusFromJson()
         }
     }
 }
+
+std::string Worker::getPrimaryHwPath()
+{
+    if (m_parsedJson.contains("backupAndRestore"))
+    {
+        return m_parsedJson["backupAndRestore"].value("PrimaryHardwarePath",
+                                                      "");
+    }
+    return std::string{};
+}
+
+std::string Worker::getSecondaryHwPath()
+{
+    if (m_parsedJson.contains("backupAndRestore"))
+    {
+        return m_parsedJson["backupAndRestore"].value("SecondaryHardwarePath",
+                                                      "");
+    }
+    return std::string{};
+}
+
+types::BackupMapVariant Worker::getBackupMap()
+{
+    if (m_parsedJson.contains("backupAndRestore"))
+    {
+        const std::string& l_backupMap =
+            m_parsedJson["backupAndRestore"].value("BackupMap", "");
+        if (l_backupMap.find("IBMSystemBackupMap") != std::string::npos)
+        {
+            return utils::IBMSystemBackupMap;
+        }
+    }
+    return std::monostate{};
+}
+
+void Worker::backupAndRestore(types::VPDMapVariant& io_parsedVpdMap)
+{
+    logging::logMessage("BackupAndRestore is Started.");
+    const std::string& l_primaryHwPath = getPrimaryHwPath();
+    const std::string& l_secondaryHwPath = getSecondaryHwPath();
+
+    if (l_primaryHwPath.empty() && l_secondaryHwPath.empty())
```

Comment:
These checks are at 3 different places in the function. Can be optimized.

Reply:
This is Initial check for both primary and secondary can't be cache.
and after this either primary or secondary paths can be given, so need a check before using .

Reply:
I don't see it can be further optimised.